connectors/sources/sharepoint_online.py

```diff
@@ -748,6 +798,80 @@ def _prefix_email(email):
     return _prefix_identity("email", email)
 
 
+def _postfix_group(group):
+    if group is None:
+        return None
+
+    return f"{group} Members"
+
+
+def is_domain_group(user_fields):
+    return user_fields["ContentType"] == "DomainGroup"
+
+
+def is_person(user_fields):
+    return user_fields["ContentType"] == "Person"
+
+
+def _domain_group_id(user_info_name):
+    """Extracts the domain group id.
+
+    The domain group id can have the following formats:
+    - abc|def|domain-group-id
+    - abc|def|some-prefix/domain-group-id
+
+    Returns:
+        str: domain group id
+
+    """
+    if user_info_name is None or len(user_info_name) == 0:
+        return None
+
+    name_parts = user_info_name.split("|")
+
+    if len(name_parts) <= 2:
+        return None
+
+    domain_group_id = name_parts[2]
+
+    if len(domain_group_id) == 0:
+        return None
+
+    if "/" in domain_group_id:
+        domain_group_id = domain_group_id.split("/")[1]
+
+    return domain_group_id
+
+
+async def _emails_and_usernames_of_domain_group(
+    domain_group_id, group_identities_generator, only_usernames=False, prefix=True
+):
```

Comment:
This method specifically looks like instance-specific cause `generator` is actually doing calls to server. I think this method should not be separated from the instance who uses it so much.

Reply:
Per definition it's not instance-specific as it misses the `self` reference, but agree that it looks like one. I can also move it down again, if you prefer so.